### DISCUSSION TITLE
AdGuard features question

### DISCUSSION THREAD

**Discussion by AseKarlsson** [AdguardTeam/AdGuardHome#6892] [Q&A] [answered]
Hi All,

Have not yet installed this on my OPNsense.

The idea is to replace ZenArmor free and use something else, this because lot of the nice stuff in ZenArmor is in the paid version and that is quite expensive.

Here a couple of questions before I try it out:
-	Can I block stuff in groups like Media Streaming (and detailed selections like Youtube, Tiktok etc.), not only Adult, violence etc.?
-	Can I exclude internal IPs/Hosts from hitting these rules?
-	Can AdGuard run together with ZenArmor Free to get benefits from both products?

Thanks,
Ase

**Comment by ghost** [comment]
Hi there!

1. Sure can!
2. Absolutely!
3. Yup!

As a note, be aware that the more layers you add to network filtering, the more complexities will arise.

**Reply by AseKarlsson** [reply]
Sounds great, will give it a try!
Any good guides to recommend for the setup and use?

**Reply by ghost** [reply, marked as answer]
[This](https://0x2142.com/how-to-set-up-adguard-on-opnsense/) one seems fairly up to date from what I read if installing it ON OPNsense is required.

**Reply by AseKarlsson** [reply]
Thanks!

**Reply by AseKarlsson** [reply]
Now Up and Running with ADGuard on my OPNsense, but have a question to be able to Unblock certain clients from the rules.

I have a domain controller running internal DNS, so ADGuard sees all DNS requests coming from the same IP (the domain controller) and not the actual clients.
Anything to do here?

Thanks,
Ase
